packages/vats/src/provisionPool.js

```diff
@@ -0,0 +1,159 @@
+  fit(privateArgs, privateShape, 'provisionPool privateArgs');
+  const { poolBank } = privateArgs;
+  fit(perAccountInitialAmount, AmountShape, 'perAccountInitialAmount');
+  const { brand: poolBrand } = perAccountInitialAmount;
+  /** @type {ERef<Purse>} */
+  // @ts-expect-error vbank purse is close enough for our use.
+  const fundPurse = E(poolBank).getPurse(poolBrand);
+
+  const zoe = zcf.getZoeService();
+  const swap = async (payIn, amount, instance) => {
+    const psmPub = E(zoe).getPublicFacet(instance);
+    const proposal = harden({ give: { In: amount } });
+    const invitation = E(psmPub).makeWantMintedInvitation();
+    const seat = E(zoe).offer(invitation, proposal, { In: payIn });
+    const payout = await E(seat).getPayout('Out');
+    return E(fundPurse).deposit(payout);
+  };
+
+  /** @type {MapStore<Brand, Instance>} */
+  const brandToPSM = makeScalarMapStore();
+
+  observeIteration(E(poolBank).getAssetSubscription(), {
+    updateState: async desc => {
+      console.log('provisionPool notified of new asset', desc.brand);
+      await zcf.saveIssuer(desc.issuer, desc.issuerName);
+      /** @type {ERef<Purse>} */
+      // @ts-expect-error vbank purse is close enough for our use.
+      const exchangePurse = E(poolBank).getPurse(desc.brand);
+      observeNotifier(E(exchangePurse).getCurrentAmountNotifier(), {
+        updateState: async amount => {
+          console.log('provisionPool balance update', amount);
+          if (AmountMath.isEmpty(amount) || amount.brand === poolBrand) {
+            return;
+          }
+          if (!brandToPSM.has(desc.brand)) {
+            console.error('funds arrived but no PSM instance', desc.brand);
+            return;
+          }
+          const instance = brandToPSM.get(desc.brand);
+          const payment = E(exchangePurse).withdraw(amount);
+          await swap(payment, amount, instance).catch(async reason => {
+            console.error(X`swap failed: ${reason}`);
+            const resolvedPayment = await payment;
+            E(exchangePurse).deposit(resolvedPayment);
+          });
+        },
+        fail: reason => console.error(reason),
+      });
+    },
+  });
+
+  const sendInitialPayment = async (address, depositFacet) => {
+    console.log('sendInitialPayment', address);
+    const initialPmt = await E(fundPurse).withdraw(perAccountInitialAmount);
+
+    return E(depositFacet)
+      .receive(initialPmt)
+      .then(amt => {
+        console.log('provisionPool sent', amt, 'to', address);
+      })
+      .catch(e => {
+        console.error(X`initial deposit for ${q(address)} failed: ${q(e)}`);
+        E(fundPurse).deposit(initialPmt);
+      });
+  };
+
+  const creatorFacet = Far('Provisioning Pool creator', {
+    makeHandler: makeBridgeProvisionTool(sendInitialPayment),
+    initPSM: (brand, instance) => {
+      fit(brand, M.remotable('brand'));
+      fit(instance, M.remotable('instance'));
+      brandToPSM.init(brand, instance);
+    },
+  });
```

Comment:
Using `makeHeapFarInstance()` moves the parameter checking into the interface declaration.  It's a few more lines, but not more complex. Your choice whether to switch or not.
```suggestion
  const creatorFacetI = M.makeInstance('provisionPool creator Facet', {
    makeHandler: M.call().returns(M.any()), // I didn't try to figure out this type.
    initPSM: M.call(BrandShape, InstanceHandleShape),
  });
  
  const creatorFacet = makeHeapFarInstance(
    'Provisioning Pool creator',
    creatorFacetI,
    {
    makeHandler: makeBridgeProvisionTool(sendInitialPayment),
    initPSM: (brand, instance) => {
      brandToPSM.init(brand, instance);
    },
  });
```

Reply:
tried that. `M.makeInstance()` doesn't seem to be defined. I tried `M.instance()` but got:

```
2022-09-11T20:45:28.396Z SwingSet: vat: v1: walletFactoryStartResult settled; remaining: [ 'client', 'provisionPoolStartResult' ]
2022-09-11T20:45:29.607Z SwingSet: ls: v18: Logging sent error stack (Error#1)
2022-09-11T20:45:29.608Z SwingSet: ls: v18: Error#1: unrecognize method guard { optional: [Function: optional], rest: [Function: rest], returns: [Function: returns] }
2022-09-11T20:45:29.608Z SwingSet: ls: v18: Error: unrecognize method guard (an object)
 at construct ()
 at Error (/home/connolly/projects/agoric-sdk/packages/SwingSet/src/supervisors/subprocess-xsnap/lockdown-subprocess-xsnap.js:6160)
 at makeError (/home/connolly/projects/agoric-sdk/packages/SwingSet/src/supervisors/subprocess-xsnap/lockdown-subprocess-xsnap.js:2717)
 at fail (/home/connolly/projects/agoric-sdk/packages/SwingSet/src/supervisors/subprocess-xsnap/lockdown-subprocess-xsnap.js:2844)
 at baseAssert (/home/connolly/projects/agoric-sdk/packages/SwingSet/src/supervisors/subprocess-xsnap/lockdown-subprocess-xsnap.js:2862)
 at interface (.../store/src/patterns/patternMatchers.js:1655)
 at start (.../vats/src/provisionPool.js:158)
 at startContract (.../zoe/src/contractFacet/zcfZygote.js:332)
```

Reply:
Sorry, that should have been `M.interface()`.   Not crucial for this pass.